projects/optic/src/commands/capture/capture.ts

```diff
@@ -131,6 +139,31 @@ const getCaptureAction =
       return;
     }
 
+    if (options.init) {
+      // no optic.yml is present but the command would attempt to write
+      if (!config.configPath && !options.stdout) {
+        logger.info(
+          'This command would write to your optic.yml, but none was found. If you want to generate an optic.yml in this directory, run `optic beta capture openapi.yml --init --stdout > optic.yml`.'
+        );
+        return;
+      }
+
+      // don't write a capture block when it would be destructive
+      if (
+        config.capture &&
+        Object.keys(config.capture).length !== 0 &&
+        !options.stdout
+      ) {
+        logger.info(
+          'Your optic.yml already contains a `capture` block with contents and this command would modify it. Make modifications to your capture config by editing optic.yml. If you want to see a complete capture block example, run `optic beta capture openapi.yml --init --stdout` instead.'
+        );
+        return;
+      }
+
+      initCaptureConfig(filePath, options.stdout!, config.configPath!);
```

Comment:
So what I think we should do here instead is:
- If there's no file, we write a file to `config.root`
- If there is a no capture key - we write a `capture:` key
- If there's an existing key that matches the openapi file name
  - we log a warnning
- If not, we can inject the file here

Not entirely sure what functions we can use to inject into a yml doc and preserve comments but we can pair on this

Reply:
> If there's an existing key that matches the openapi file name
> we log a warning and log out instead
> If not, we can inject the capture config with the file name

i don't know that i agree that its important for this to support inserting more than 1 capture[oas_file] section. once you have the first one, you're bootstrapped—you've seen the complete example already. adding more capture configuration manually is trivial at that point, right?

Reply:
> If there's no file, we write a file to config.root
> If there is a no capture key - we write a capture: key

these are handled

Reply:
You're right that we're likely going to use this for the the first use for the majority of the time, but I don't think it unreasonable for someone to run capture init when they've already added a capture config.

It's probably not that important to have for the first launch of this but I think it does add a level of polish here